### PR TITLE
planning: improve efficiency for TrajectoryEvaluator.

### PR DESCRIPTION
**Pull request by davidhopper2003**

### CODE REVIEW

**Comment by startcode** [comment]
please run ./apollo.sh check and fix the failed lint tests.
Thanks.

File: modules/planning/lattice/trajectory_generation/trajectory_evaluator.cc [hunk 1]
```diff
@@ -82,12 +82,13 @@ TrajectoryEvaluator::TrajectoryEvaluator(
...
         double cost = Evaluate(planning_target, lon_trajectory, lat_trajectory);
-        cost_queue_.push(PairCost({lon_trajectory, lat_trajectory}, cost));
+        // Use more efficient "emplace" member fucntion. 
+        cost_queue_.emplace(PairCost({lon_trajectory, lat_trajectory}, cost));
       } else {
         std::vector<double> cost_components;
         double cost = Evaluate(planning_target, lon_trajectory, lat_trajectory,
                                &cost_components);
-        cost_queue_with_components_.push(PairCostWithComponents(
+        cost_queue_with_components_.emplace(PairCostWithComponents(
```
Comment (lianglia-apollo) [inline modules/planning/lattice/trajectory_generation/trajectory_evaluator.cc]: this is not necessary. push or emplace here have no difference.
It is generally preferred to keeping using push if there is no performance improvement.

File: modules/planning/lattice/trajectory_generation/trajectory_evaluator.cc [hunk 2]
```diff
@@ -173,7 +174,7 @@ double TrajectoryEvaluator::Evaluate(
   std::vector<double> s_values;
   for (double s = 0.0; s < evaluation_horizon;
        s += FLAGS_trajectory_space_resolution) {
-    s_values.push_back(s);
+    s_values.emplace_back(s);
```
Comment (lianglia-apollo) [inline modules/planning/lattice/trajectory_generation/trajectory_evaluator.cc]: this is not necessary. push or emplace here have no difference.

Comment (davidhopper2003) [reply modules/planning/lattice/trajectory_generation/trajectory_evaluator.cc]: For native data types, the “emplace_back” or "emplace" function doesn't improve efficiency over the "push_back" or "push" fucntion; but for non-native data types, the formmer doesn't invoke an extra constructor and thus the efficiency is enhanced. Whether for native or non-native data types, the C++11 standard recommends using the "emplace_back" function. Therefore, I think my change is no problem.

**Comment by davidhopper2003** [comment]
@startcode Yes, I ignored the code style checking test. I will finish this job at once.

File: modules/planning/lattice/trajectory_generation/trajectory_evaluator.cc (hunk 1 above)
Comment (lianglia-apollo) [reply modules/planning/lattice/trajectory_generation/trajectory_evaluator.cc]: Please use cost_queue_with_components_.emplace(lon_trajectory, lat_trajectory).

The way you wrote (emplace(PairCostWithComponents{xxx, xxx}) requires a construction of temporary instance and invokes the copy constructor, which is identical to what you will get with a push().

Comment (davidhopper2003) [reply modules/planning/lattice/trajectory_generation/trajectory_evaluator.cc]: Great, thank you for pointing out my problems. I will correct them right away.

Comment (davidhopper2003) [reply modules/planning/lattice/trajectory_generation/trajectory_evaluator.cc]: I have modified the code, please check again. You are awesome!

**Pull request merged**